sdk/synapse/synapse-artifacts/review/synapse-artifacts.api.md

```diff
@@ -0,0 +1,5227 @@
+    readonly output?: any;
+    readonly pipelineName?: string;
+    readonly pipelineRunId?: string;
+    readonly status?: string;
+}
+
+// @public
+export interface ActivityRunsQueryResponse {
+    continuationToken?: string;
+    value: ActivityRun[];
+}
+
+// @public (undocumented)
+export type ActivityUnion = ControlActivity | ExecutionActivityUnion | ExecutePipelineActivity | IfConditionActivity | SwitchActivity | ForEachActivity | WaitActivity | UntilActivity | ValidationActivity | FilterActivity | SetVariableActivity | AppendVariableActivity | WebHookActivity | SynapseNotebookActivity | SynapseSparkJobDefinitionActivity | SqlPoolStoredProcedureActivity;
+
+// @public
+export type AddDataFlowResponse = AddDataFlowToDebugSessionResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: AddDataFlowToDebugSessionResponse;
+    };
+};
+
+// @public
+export interface AddDataFlowToDebugSessionResponse {
+    jobVersion?: string;
+}
+
+// @public
+export type AmazonMWSLinkedService = LinkedService & {
+    endpoint: any;
+    marketplaceID: any;
+    sellerID: any;
+    mwsAuthToken?: SecretBaseUnion;
+    accessKeyId: any;
+    secretKey?: SecretBaseUnion;
+    useEncryptedEndpoints?: any;
+    useHostVerification?: any;
+    usePeerVerification?: any;
+    encryptedCredential?: any;
+};
+
+// @public
+export type AmazonMWSObjectDataset = Dataset & {
+    tableName?: any;
+};
+
+// @public
+export type AmazonMWSSource = TabularSource & {
+    query?: any;
+};
+
+// @public
+export type AmazonRedshiftLinkedService = LinkedService & {
+    server: any;
+    username?: any;
+    password?: SecretBaseUnion;
+    database: any;
+    port?: any;
+    encryptedCredential?: any;
+};
+
+// @public
+export type AmazonRedshiftSource = TabularSource & {
+    query?: any;
+    redshiftUnloadSettings?: RedshiftUnloadSettings;
+};
+
+// @public
+export type AmazonRedshiftTableDataset = Dataset & {
+    tableName?: any;
+    table?: any;
+    schemaTypePropertiesSchema?: any;
+};
+
+// @public
+export type AmazonS3LinkedService = LinkedService & {
+    accessKeyId?: any;
+    secretAccessKey?: SecretBaseUnion;
+    serviceUrl?: any;
+    encryptedCredential?: any;
+};
+
+// @public
+export type AmazonS3Location = DatasetLocation & {
+    bucketName?: any;
+    version?: any;
+};
+
+// @public
+export type AmazonS3ReadSettings = StoreReadSettings & {
+    recursive?: any;
+    wildcardFolderPath?: any;
+    wildcardFileName?: any;
+    prefix?: any;
+    enablePartitionDiscovery?: boolean;
+    modifiedDatetimeStart?: any;
+    modifiedDatetimeEnd?: any;
+};
+
+// @public
+export type AppendVariableActivity = Activity & {
+    variableName?: string;
+    value?: any;
+};
+
+// Warning: (ae-forgotten-export) The symbol "AuthenticationClient" needs to be exported by the entry point index.d.ts
+//
+// @public (undocumented)
+export class ArtifactsClient extends AuthenticationClient {
```

Comment:
why are we using inheritance for the various clients? Is it simply to re-use setting up the pipeline? Could we instead extract whatever is in the constructor for AuthenticationClient and put it in a common place?